lib/Sema/CSSimplify.cpp

```diff
@@ -8200,6 +8200,22 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyConformsToConstraint(
     return SolutionKind::Solved;
   }
 
+  // Copyable is checked structurally, so for better performance, split apart
+  // this constraint into individual Copyable constraints on each tuple element.
+  if (auto *tupleType = type->getAs<TupleType>()) {
+    if (protocol->isSpecificProtocol(KnownProtocolKind::Copyable)) {
+      for (unsigned i = 0, e = tupleType->getNumElements(); i < e; ++i) {
+        auto elm = tupleType->getElement(i);
+        addConstraint(ConstraintKind::ConformsTo,
+                      elm.getType(),
```

Comment:
Nit: you can just use `tupleType->getElementType(i)`

Reply:
ah thanks!